iOSDevUK/Controllers/Sessions/ProgrammeTableViewController.swift

```diff
@@ -86,6 +98,41 @@ class ProgrammeTableViewController: UIViewController, UITableViewDelegate, UITab
         }
     }
     
+    /// Find the current or next session for the current set of sessions and scroll to it.
+    func scrollToCurrentSession(animated: Bool) {
+        guard let fetchedResultsController = fetchedResultsController,
+                let sections = fetchedResultsController.sections else {
+            return
+        }
+        
+        /// Find the session in the already-fetched results
+        let now = Date()
+        let nextOrCurrentSession = fetchedResultsController.fetchedObjects?.first { session in
+            guard let startTimeNSDate = session.startTime,
+                    let endTimeNSDate = session.endTime else {
+                return false
+            }
+            let startTime = startTimeNSDate as Date
+            let endTime = endTimeNSDate as Date
+            return (startTime <= now && endTime >= now) || (startTime >= now)
+        }
+        
+        /// If we found one, resolve it to a section index and row
```

Comment:
You can ask the fetchedResultsController for an indexPath directly, so you don't need to search for it yourself:

```swift
let indexPathOfSessionToShow = fetchedResultsController.indexPath(forObject: nextOrCurrentSession)
tableView.scrollToRow(at: indexPathOfSessionToShow, at: .top, animated: animated)
```